plugins/pluginmanagers/lazy.nix

```diff
@@ -1,222 +1,264 @@
+                      opts = {
+                        load."['core.defaults']" = [ ];
+                      };
+                    }
+                  ];
+                  ```
+                '';
+
+                keys = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on key mapping";
+
+                module = mkNullOrOption (enum [ false ]) ''
+                  Do not automatically load this Lua module when it's required somewhere.
+                '';
+
+                priority = defaultNullOpts.mkInt 50 ''
+                  Only useful for start plugins i.e.
+                  `plugins.lazy.plugins.<plugin>.lazy = false` to force loading
+                  certain plugins first. Default priority is 50. It's
+                  recommended to set this to a high number for colorschemes.
+                '';
+
+                optional = defaultNullOpts.mkBool false ''
+                  Optional specs are only included in the final configuration
+                  if the corresponding plugin is also specified as a required
+                  (non-optional) plugin elsewhere. This feature is
+                  particularly helpful for Neovim distributions, allowing
+                  them to pre-configure settings for plugins that users may
+                  or may not have installed.
+                '';
+
+                opts = mkNullOrOption (maybeRaw (attrsOf anything)) ''
+                  The opts value can be one of the following:
+
+                  - A table: This table will be merged with any existing
+                  configuration settings from parent specifications.
+                  - A function that returns a table: The returned table will
+                  completely replace any existing configuration settings from
+                  parent specifications.
+                  - A function that modifies a table: This function will
+                  receive the existing configuration table as an argument and
+                  can modify it directly.
+
+                  In all cases, the resulting configuration table will be
+                  passed to the Plugin.config() function. Setting the opts
+                  value automatically implies that Plugin.config() will be
+                  called. (See: https://lazy.folke.io/spec#spec-setup)
+                '';
+              };
+
+              config.name = lib.mkIf (cfg.pkg != null) (lib.mkDefault "${lib.getName cfg.pkg}");
+              config.dir = lib.mkIf (cfg.pkg != null) (lib.mkDefault "${cfg.pkg}");
+            }
+          )
+        );
+
+      lazyPluginsListType = types.listOf lazyPluginType;
+    in
+    {
+      gitPackage = lib.mkPackageOption pkgs "git" { nullable = true; };
+      luarocksPackage = lib.mkPackageOption pkgs "luarocks" { nullable = true; };
+
+      plugins = lib.mkOption {
+        type = lazyPluginsListType;
+        default = [ ];
+        description = "List of plugins";
+      };
+    };
```

Comment:
This option would benefit greatly from an `example` and also a fleshed out `description`.